### PR TITLE
fvm: init at 3.2.1

### PR DESCRIPTION
**Pull request by MH0386**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fv/fvm/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  lib,
+}:
+
+buildDartApplication rec {
+  pname = "fvm";
+  version = "3.2.1";
+  src = fetchFromGitHub {
+    owner = "leoafarias";
+    repo = pname;
+    rev = "refs/tags/${version}";
```
Comment (lucasew) [inline pkgs/by-name/fv/fvm/package.nix:18]: `tag = version;`

File: pkgs/by-name/fv/fvm/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  pkg-config,
+  gtk3,
+  cmake,
+  fetchFromGitHub,
+  lib,
+}:
+
+buildDartApplication rec {
+  pname = "fvm";
+  version = "3.2.1";
```
Comment (lucasew) [inline pkgs/by-name/fv/fvm/package.nix:14]: please add a empty line between version and src

File: pkgs/by-name/fv/fvm/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  '';
+  dontUseCmakeConfigure = true;
+  postInstall = ''
+    export HOME=$TMPDIR
+    installShellCompletion --cmd fvm \
+      --bash <($out/bin/fvm --generate-shell-completion bash) \
+      --fish <($out/bin/fvm --generate-shell-completion fish) \
+      --zsh <($out/bin/fvm --generate-shell-completion zsh)
+  '';
+  meta = with lib; {
```
Comment (lucasew) [inline pkgs/by-name/fv/fvm/package.nix:42]: Please use values directly in meta, without the `with lib`.

File: pkgs/by-name/fv/fvm/package.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    description = "Flutter Version Management: A simple CLI to manage Flutter SDK versions.";
+    homepage = "https://fvm.app";
+    license = licenses.mit;
+    platforms = platforms.all;
+    # maintainers = with maintainers; [  ];
```
Comment (lucasew) [inline pkgs/by-name/fv/fvm/package.nix:47]: Please add yourself as a maintainer. If your name is not in the list please add a commit that adds your name into `maintainers/maintainer-list.nix`.

**Review by lucasew — CHANGES_REQUESTED** [review]
Also please rename the PR and the package commit to `fvm: init at 3.2.1`. This comes from CONTRIBUTING.md.

File: pkgs/by-name/fv/fvm/package.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    export LD_LIBRARY_PATH=${gtk3.out}/lib:$LD_LIBRARY_PATH
+  '';
+  dontUseCmakeConfigure = true;
+  postInstall = ''
+    export HOME=$TMPDIR
+    installShellCompletion --cmd fvm \
+      --bash <($out/bin/fvm --generate-shell-completion bash) \
+      --fish <($out/bin/fvm --generate-shell-completion fish) \
+      --zsh <($out/bin/fvm --generate-shell-completion zsh)
+  '';
```
Comment (lucasew) [inline pkgs/by-name/fv/fvm/package.nix:41]: `passthru.updateScript = gitUpdater { };` to add automatic updates

File: pkgs/by-name/fv/fvm/package.nix [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+buildDartApplication rec {
+  pname = "fvm";
+  version = "3.2.1";
+  src = fetchFromGitHub {
+    owner = "leoafarias";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    hash = "sha256-i7sJRBrS5qyW8uGlx+zg+wDxsxgmolTMcikHyOzv3Bs=";
+  };
+  autoPubspecLock = "${src}/pubspec.lock";
```
Comment (lucasew) [inline pkgs/by-name/fv/fvm/package.nix:21]: This, AFAIK, is not allowed. That's called import from derivation and is basically when the eval step depends on a build step and this is not allowed in nixpkgs.

CI is also complaining that the file is not there.

IDK how it is the state now but in packages I mantained previously I had to save the lockfile converted to json along with the package.nix.

File: pkgs/by-name/fv/fvm/package.nix (hunk 4 above)
Comment (MH0386) [reply pkgs/by-name/fv/fvm/package.nix:48]: in the same branch?

Comment (lucasew) [reply pkgs/by-name/fv/fvm/package.nix]: Yes. In the end there should be two commits. One that adds your name to the maintainer list and another with all the rest.

File: pkgs/by-name/fv/fvm/package.nix (hunk 6 above)
Comment (MH0386) [reply pkgs/by-name/fv/fvm/package.nix]: Why I cannot? I wanted to use the lock file from the repo to be up-to-date.

Comment (lucasew) [reply pkgs/by-name/fv/fvm/package.nix]: Nixpkgs assumes one would evaluate all nix code, generate all the derivations then start building.

With this, the eval phase would depend on a build and this is forbidden, and make stuff much slower.

**Comment by MH0386** [comment]
I curious about `ofborg`. Why it takes long time?
@lucasew

**Comment by lucasew** [comment]
> I curious about `ofborg`. Why it takes long time? @lucasew

A new commit restarts the process. And stuff is processed in a queue where there is other PRs.

**Comment by MH0386** [comment]
What next?

**Comment by lucasew** [comment]
Join (squash commits).

One to add your name in the maintainers list and another to add your package.

**Comment by K900** [comment]
What the hell is happening here. Please clean up your commit history and don't touch unrelated files.

**Pull request closed without merging**